kahuna/public/js/routes/index.js

```diff
@@ -0,0 +1,45 @@
+import angular from 'angular';
+import {searchQueryService} from '../search-query/service';
+
+// Use the name once that PR in merged in
+import '../services/api/media-api';
+
+export const indexRoute = angular.module('gr.routes.index', []);
+
+indexRoute.config(['$stateProvider', '$urlMatcherFactoryProvider',
+    function($stateProvider, $urlMatcherFactoryProvider) {
+
+        const zeroWidthSpace = /[\u200B]/g;
+        function removeUtf8SpecialChars(val) {
+            return angular.isDefined(val) && val.replace(zeroWidthSpace, '');
+        }
+        $urlMatcherFactoryProvider.type('Query', {
+            encode: val => removeUtf8SpecialChars(val),
+            decode: val => removeUtf8SpecialChars(val),
+            //call decode value that includes zero-width-space character
+            is: val => angular.isDefined(val) && !zeroWidthSpace.test(val)
+        });
+
+        $stateProvider.state('gr', {
+            url: '?{query:Query}&ids&since&nonFree&uploadedBy&until&orderBy',
+            abstract: true,
+            template: `<ui-view />`,
+            resolve: {
+                searchResults$: ['searchQueryService', 'mediaApi', function(searchQueryService, mediaApi) {
```

Comment:
The more I work with this, the more I think that these shouldn't be in the route resolvers, but i there own state services.

The reason being - the dependency is then available to the route controller, but any other controller within that can't include it as a dependency.

A good example is the [`selectedImages$`](https://github.com/guardian/grid/blob/master/kahuna/public/js/search/index.js#L142), which [child views of `search.results`](https://github.com/guardian/grid/blob/master/kahuna/public/js/search/index.js#L161-L229) [can access](https://github.com/guardian/grid/blob/master/kahuna/public/js/search/index.js#L161-L229) but controllers within that can't, so you have to pass it down in a [chain like fashion](https://github.com/guardian/grid/blob/master/kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js#L150). 

I think what would be more beneficial is have a `app-state/selected-images` service, which could then be used from across the app.




https://github.com/guardian/grid/blob/master/kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js#L60-L61